target_s3_jsonl/s3.py

```diff
@@ -43,10 +48,20 @@ def create_client(config):
     else:
         aws_session = boto3.session.Session(profile_name=aws_profile)
 
-    if aws_endpoint_url:
-        return aws_session.client('s3', endpoint_url=aws_endpoint_url)
-    else:
-        return aws_session.client('s3')
+    # config specifies a particular role to assume
+    # we create a session & s3-client with this role
+    if role_arn:
+        role_name = role_arn.split('/', 1)[1]
+        sts = aws_session.client('sts', **endpoint_params)
+        resp = sts.assume_role(RoleArn=role_arn, RoleSessionName=f'Meltano-role={role_name}-profile={aws_profile}')
```

Comment:
I'm not sure `Meltano-role=` should be hard coded there as this target shall be used by people outside meltano.

Reply:
that's a good point. Do you think it would suffice to set `RoleSessionName=f'role-name={role_name}-profile={aws_profile}'`